### PR TITLE
Sync upstream

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
Sync to `2d2cc76ffb781d01c800608cd8be05cca67ac4c0`.

### CODE REVIEW

**Comment by shayne-fletcher-da** [comment]
Current set of failures all seem to be network related.

**Comment by shayne-fletcher-da** [comment]
The change is good. CI is looking a bit broke - network issues affecting linux and Windows.

File: ghc-lib-gen/src/Ghclibgen.hs [hunk 1]
```diff
@@ -208,6 +208,7 @@ cHeaders :: GhcFlavor -> [String]
 cHeaders ghcFlavor =
   [ "includes/ghcconfig.h"
   , "includes/MachDeps.h"
+  , "includes/stg/MachRegs.h"
```
Comment (cocreature) [inline ghc-lib-gen/src/Ghclibgen.hs]: Shouldn’t this change depend on the STG version? I don’t see how this could have changed for older versions.

Comment (shayne-fletcher-da) [reply ghc-lib-gen/src/Ghclibgen.hs]: Not an oversight. The inclusion of a new set of constants to support the LLVM backend highlighted a bug in our inclusion scheme. The contents of `includes/CodeGen.Platform.hs` includes ` stg/MachRegs.h`. The former is in the `extra-sources` of ghc-lib targets but not the latter which we've been satisfying by provision from the compiler. Now we include `MachRegs.h` (as we should) along with `MachDeps.h`.

Comment (cocreature) [reply ghc-lib-gen/src/Ghclibgen.hs]: Sounds good :+1:

**Pull request merged**